imx: fix race condition in interrupt handling

iuliana-prodan opened the pull request:
See each commit message

dbaluta reviewed (changes requested):
Let's wait for more testing. I have a different patch for this.

iuliana-prodan commented:
> Let's wait for more testing. I have a different patch for this.

@dbaluta this was tested on 2 different 8QM boards with both SRC and MIXER tplgs, with suspend-resume.
The tests were run 5 times in a row, with no failure or crash.

iuliana-prodan commented:
I've added a 3rd commit to this PR to fix qemu boot failure on imx8 (this was due to commit 1 in which I've updated the interrupt initialization sequence).

dbaluta commented on src/drivers/imx/ipc.c:
```diff
@@ -69,7 +69,6 @@ static void irq_handler(void *arg)
 	/* new message from host */
 	if (status & IMX_MU_xSR_GIPn(IMX_MU_VERSION, 0)) {
-
 		/* Disable GP interrupt #0 */
```
this seems unrelated. Is this intentional?

dbaluta commented:
Overnight tests passed. Excellent work Iulia!

iuliana-prodan commented:
> Overnight tests passed. Excellent work Iulia!

Thanks! It was a joint work!
